pkg/action/executor/move_util.go

```diff
@@ -97,22 +121,76 @@ func movePod(client *kclient.Clientset, pod *api.Pod, nodeName string, retryNum
 		}
 	}()
 
-	//2 wait until podC gets ready
-	err = podutil.WaitForPodReady(client, npod.Namespace, npod.Name, nodeName,
+	// Special handling for pods with volumes
+	if podUsingVolume {
+		// We still support replicaset and replication controllers as parents,
+		// however both of them could further have a parent (deploy or deploy config).
+		parent, grandParent, pClient, gPClient, gPKind, err :=
+			getParentAndGrandParentInfo(clusterScraper, pod, parentKind)
+		if err != nil {
+			return nil, err
+		}
+
+		if grandParent != nil {
+			pause := true
+			unpause := false
+			// step B8: (via defer)
+			defer func() {
+				glog.V(3).Infof("Unpausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
+				resourceRollout(gPClient, grandParent, unpause)
+			}()
+
+			// step B2:
+			glog.V(3).Infof("Pausing pods controller: %s for pod: %s", gPKind, podQualifiedName)
+			err := resourceRollout(gPClient, grandParent, pause)
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		valid := true
+		invalid := false
+		// step B7: (via defer)
+		defer func() {
+			glog.V(3).Infof("Updating scheduler of %s for pod %s back to default.", parentKind, podQualifiedName)
+			changeScheduler(pClient, parent, valid)
+
+		}()
+		// step B3:
+		glog.V(3).Infof("Updating scheduler of %s for pod %s to unknown (turbo-scheduler).", parentKind, pod.Name)
+		changeScheduler(pClient, parent, invalid)
```

Comment:
Do we need to check error here and return if there is anything wrong?

Reply:
Thanks for catching this. Updated!